lib/rules/require-expect.js

```diff
@@ -67,10 +76,16 @@ module.exports = {
         }
 
         function isPassingAssertAsArgument(node) {
+            const allowedFunctions = context.options[1] && context.options[1].allowFunctions || [];
+
             if (!currentTest.assertName) {
                 return false;
             }
 
+            if (allowedFunctions.includes(node.callee.name)) {
```

Comment:
This will only work with identifiers. Do we want to support property accesses (for example, `myLib.preprocess(assert)`?